numpy/lib/npyio.py

```diff
@@ -86,10 +86,12 @@ def __dir__(self):
         return object.__getattribute__(self, '_obj').keys()
 
 
-def zipfile_factory(*args, **kwargs):
+def zipfile_factory(file, mode='r', *args, **kwargs):
```

Comment:
Just to clarify the procedure ;) A new private function, say `_zipfile_factory` would be defined, and `zipfile_factory` would be deprecated. That may seem overly picky, but all such changes carry some risk, and when several hundred PRs are present in a release the risk accumulates.